src/manual/using/actions/ui.md

```diff
@@ -30,24 +30,42 @@ For `action.kind = 'task'`:
     user's Taskcluster credentials. See the next section for some
     important security-related concerns.
 
+For `action.kind = 'hook'`:
+
+-   Rendering the hookPayload using JSON-e with the context describe
+    in the specification.
+-   Calling `Hooks.triggerHook` with the resulting hook identiifers
+    and payload.
+
 User interfaces should ignore actions with unrecognized kinds.
 
-Creating Tasks
---------------
+Security Concerns
+-----------------
 
-When executing an action, a UI must ensure that the user is authorized
-to perform the action, and that the user is not being "tricked" into
-executing an unexpected action.
+When executing an action, a UI must ensure that the user is authorized to
+perform the action, and that the user is not being "tricked" into executing an
+unexpected action. The `actions.json` artifact should be treated as untrusted
+content!
 
 To accomplish the first, the UI should create tasks with the user's
 Taskcluster credentials. Do not use credentials configured as part of
 the service itself!
 
-To accomplish the second, use the decision tasks' `scopes` property as
-the
+### "task" Actions
+
+To accomplish the second for "task" actions, use the decision tasks' `scopes`
+property as the
 [authorizedScopes](https://docs.taskcluster.net/manual/design/apis/hawk/authorized-scopes)
-for the `Queue.createTask` call. This prevents action tasks from doing
-anything the original decision task couldn't do.
+for the `Queue.createTask` call. This prevents action tasks from doing anything
+the original decision task couldn't do.
+
+### "hook" Actions
+
+Hook actions have no such luxury, as they are intended to do things the
+decision task couldn't do.
+
+Instead, the user interface should require the user to confirm each action,
+displaying the hookGroupId, hookId, and hookPayload.
```

Comment:
Not that we necessarily need to do this now, but I wonder if we could make this even more solid. Very poorly thought out ideas:

* require linking to that hook in the tools site?
* suggest displaying a "you have run this hook N times before" the way browsers do with ssl/website stuff
* none of the above!

Reply:
Uh, maybe require the UI display the `name`, `description` and `owner` properties of the hook.

That way a warning can be added to the description property indicating that the hook could do something bad... Stuff like that would smart..